## Carregar os dados

In [1]:
import pandas as pd

In [2]:
dadosDT = pd.read_csv('../data/testeDT.csv')
dadosNN = pd.read_csv('../data/testeNN.csv')

## Extrair uma amostra para verificar o número de acertos

In [3]:
import numpy as np

In [4]:
df_indexs = list(dadosDT.index)

In [5]:
indexs = np.random.choice(df_indexs,150,replace=False)

In [6]:
amostraDT = dadosDT.iloc[indexs]

In [7]:
amostraNN = dadosNN.iloc[indexs]

- Não executar novamente a amostragem

In [15]:
amostraNN[:10]

Unnamed: 0  tags                                               text
3089        3089     1    chupa seu porco a cana e doce e dura viu safado
4695        4695     0  bolsonaro e homofobico racista e misogino se v...
1839        1839     1  nenhum partido politico e maior do que o brasi...
2315        2315     1  alem do sonho e preciso agir criar as proprias...
3007        3007     1                                   yjcxqunlz8 15 rs
3850        3850     1                   so ler os comentarios e hilario 
4641        4641     1   pessoal o governador de mg e o prefeito da mi...
5071        5071     1   video sexo por notas cameras escondidas expoe...
1623        1623     1   come on whats with the side kick he sure does...
612          612     1  o governo bem que tenta mas a resistencia a ce...
4795        4795     1  qual o pai ou mae acha normal um presidente ca...
4098        4098     1                           foi tarde sucuri idiota 
4829        4829     1           para as mais beijinho para as recalcada 
2753        2753     1  o sonho dos esquerdopatas continua sendo so um...
1367        1367     1  fogaca juliana seu prato me surprendeu n posit...
1764        1764     1    silencio o jovem esta descobrindo o brisadeiro 
559          559     2  2 o brasil e o quarto colocado no ranking mund...
295          295     1   psol pede a pgr investigar uso de milicias di...
4617        4617     1   tudo o que o homem plantar ele mesmo colheraq...
1293        1293     1                     english speaking practice via 
493          493     1  o laranjal pode derrubar bolsonaro a censura e...
1334        1334     1  e gratificante ver a linda cia folclorica matu...
4444        4444     1                    agora sim da pra encher a cara 
4849        4849     1   odeio sair com gente q so fala de faculdade q...
3863        3863     1                                  novafotodoperfil 
3389        3389     1  rio muito com as mina do busao achando que eu ...
4009        4009     1                                             azedou
1224        1224     1   virei assinante recentemente que ap e esse us...
193          193     1  educacao educacao e educacao ideia exemplo e m...
4404        4404     1                                  eba maravilhosa  
...          ...   ...                                                ...
1706        1706     1                                meu burro favorito 
4926        4926     1   porra imagina andar na rua sem ouvir gracinha...
178          178     1  cafecomboulos entenda o que significa o acordo...
4342        4342     1                                                   
81            81     1  apos forte pressao dos motoristas de aplicativ...
5320        5320     1                ate que enfim ele falou uma verdade
2584        2584     1                     a maior decepcao para o brasil
200          200     1  participo agora da i conferencia nacional de e...
4092        4092     1   eu so nao gosto dele e acho ele forcado mas s...
3669        3669     1                  iurd igrejauniversal fjublocosul 
3681        3681     1            hj eu so quero ser mimada e comprendida
25            25     1   senado finalmente aprovou a reforma da previd...
1153        1153     1                                   com que dinheiro
419          419     1   maturidade politica por parte dos membros do ...
1239        1239     1  compra essa porcaria de livro e de de presente...
3644        3644     1    ai q ele nao consegue nada ja ta ruim pra ele k
3222        3222     1   sulamericano brasil vence colombia por 3 sets...
1606        1606     1   o clipe ta lindo a paisagem impecavel o jao j...
3090        3090     1                   tu nao eras evangelico sua besta
2189        2189     0   o maior injusticado da historia da selecao br...
1404        1404     1                                      boicote total
1756        1756     1                                      pra cima moro
1971      

In [16]:
amostraDT.to_csv('../data/amostraDT.csv')
amostraNN.to_csv('../data/amostraNN.csv')

- Após salvar os modelos foi realizado uma verificação de cada classificação indicando se a classificação estava certa e indicando a classificação correta

In [18]:
amostraDT = pd.read_csv('../data/amostraDT.csv', usecols=['tags', 'text', 'classificacao', 'pos'])

In [19]:
amostraNN = pd.read_csv('../data/amostraNN.csv', usecols=['tags', 'text', 'classificacao', 'pos'])

In [20]:
amostraNN[:10]

tags                                               text  classificacao  pos
0     1    chupa seu porco a cana e doce e dura viu safado              1    1
1     0  bolsonaro e homofobico racista e misogino se v...              1    0
2     1  nenhum partido politico e maior do que o brasi...              1    1
3     1  alem do sonho e preciso agir criar as proprias...              1    1
4     1                                   yjcxqunlz8 15 rs              1    1
5     1                   so ler os comentarios e hilario               1    1
6     1   pessoal o governador de mg e o prefeito da mi...              1    1
7     1   video sexo por notas cameras escondidas expoe...              1    1
8     1   come on whats with the side kick he sure does...              1    1
9     1  o governo bem que tenta mas a resistencia a ce...              1    1

In [21]:
estatisticaDT = amostraDT['classificacao']
estatisticaNN = amostraNN['classificacao']

In [22]:
dt = estatisticaDT.value_counts()
nn = estatisticaNN.value_counts()

In [23]:
dt

1    122
0     28
Name: classificacao, dtype: int64

In [24]:
nn

1    119
0     31
Name: classificacao, dtype: int64

- Precisão = (total de acertos)/(total de acertos + total de erros)
- DT = 122/(122+28)
- NN = 119/(119+31)

In [25]:
def precision(estatistica):
    return estatistica[1]/(estatistica[1]+estatistica[0])

In [26]:
print(precision(dt))

0.8133333333333334


In [27]:
print(precision(nn))

0.7933333333333333


# Gerar matriz de confusão

In [60]:
dfDT = amostraNN.loc[(amostraNN['pos']==2),['classificacao', 'tags']]

In [61]:
dfDT

classificacao  tags
13               0     1
44               1     2
69               1     2
83               0     1
89               0     1
93               0     1
101              0     1
103              1     2
113              0     1
117              0     1
131              0     1
141              0     1
142              0     1

In [62]:
dt = dfDT['tags'].value_counts()

In [63]:
dt

1    10
2     3
Name: tags, dtype: int64

## Verificar o acerto para posicionamento de esquerda e de direita
- remover tweets neutros

In [56]:
dfDT = amostraDT.loc[(amostraDT['pos']!=1),['classificacao', 'pos']]

In [57]:
dfNN = amostraNN.loc[(amostraNN['pos']!=1),['classificacao', 'pos']]

In [59]:
dt = dfDT['classificacao'].value_counts()
nn = dfNN['classificacao'].value_counts()

In [60]:
dt

0    24
1     7
Name: classificacao, dtype: int64

In [61]:
nn

0    24
1     7
Name: classificacao, dtype: int64

In [62]:
print(precision(dt))

0.22580645161290322


In [63]:
print(precision(nn))

0.22580645161290322
